src/index.js

```diff
@@ -2,74 +2,47 @@
  * See LICENSE for license details.
  */
-import { resolve, join } from 'path';
-import { existsSync } from 'fs-extra';
 import { info } from 'consola';
-import { log } from './helpers/log';
-import { DEFAULTS } from './helpers/constants';
-import beforeAll from './utils/beforeAll';
+import {
+    getConfigsForModules,
+    prepareOptions,
+} from './helpers/tools';
+import beforeAllModules from './utils/beforeAll';
+import prepareModules from './utils/prepareModules';
 import initModules from './utils/initModules';
-import afterAll from './utils/afterAll';
+import afterAllModules from './utils/afterAll';
 
 /**
  * A simple mechanism to transform a monolithic Vue application into an application based on Micro Services architecture.
  * @author Piotr Bletek <p.bletek@gmail.com>
  * @function VueMS Vue Micro Services
- * @param {Object} moduleOptions - Module options
+ * @param {Object} moduleOptions - VueMS initial options
  */
 export default async function VueMS(moduleOptions = {}) {
-    const options = { ...DEFAULTS, ...this.options.vuems || moduleOptions };
-    let localModules = options.modules.local || [];
-    let npmModules = options.modules.npm || [];
+    const options = prepareOptions.call(this, moduleOptions);
 
-    if (this.options.srcDir) {
-        options.vendorDir = resolve(this.options.srcDir, options.vendorDir);
-        options.modulesDir = resolve(this.options.srcDir, options.modulesDir);
-    }
-
-    if (!localModules.length && !npmModules.length) return false;
-
-    info('VueMS init');
+    if (!options) {
+        info('VueMS init - no modules to load');
 
-    if (options.modules.npm) {
-        npmModules = options.modules.npm.map(module => ({
-            name: module,
-            type: 'npm',
-            path: join(options.vendorDir, module, 'src'),
-        }));
+        return false;
     }
-    if (options.modules.local) {
-        localModules = options.modules.local.map((module) => {
-            const path = join(options.modulesDir, module, 'src');
-
-            if (existsSync(path)) {
-                return {
-                    name: module,
-                    type: 'local',
-                    path,
-                };
-            }
-
-            return {
-                name: module,
-                type: 'local',
-                path: join(options.modulesDir, module),
-            };
-        });
-    }
-
-    options.allModules = [...localModules, ...npmModules];
-
-    const beforeAllData = await beforeAll.call(this, options);
-    const initModulesMessages = await initModules.call(this, beforeAllData.modulesConfigs, options);
-    const afterAllMessages = await afterAll.call(this, options);
-    const messages = {
-        ...beforeAllData.messages,
-        ...initModulesMessages,
-        ...afterAllMessages,
-    };
+    info('VueMS init');
 
-    if (options.logProcess) {
-        log(messages);
-    }
+    await beforeAllModules.call(this, {
```

Comment:
what about naming it as beforeModules and afterModules ? Between those function calls there are called prepareModules not prepareAllModules etc.